keras_core/distribution/distribution_lib.py

```diff
@@ -288,6 +289,102 @@ def get_variable_layout(self, variable):
         return TensorLayout(variable_shard_spec, self.device_mesh)
 
 
+class LayoutMap(collections.abc.MutableMapping):
+    """A dict-like object that maps string to `TensorLayout` instances.
+
+    `LayoutMap` uses a string as key and a `TensorLayout` as value. There is a
+    behavior difference between a normal Python dict and this class. The string
+    key will be treated as a regex when retrieving the value. See the docstring
+    of `get` for more details.
+
+    See below for a usage example. You can define the naming schema
+    of the `TensorLayout`, and then retrieve the corresponding
+    `TensorLayout` instance.
+
+    In the normal case, the key to query is usually the `variable.path`, which
+    is the idenifier of the variable.
+
+    ```python
+    map = LayoutMap(device_mesh=None)
```

Comment:
`map` is a reserved keyword, so let's use something like `variable_map`

Reply:
Done.